core/app/components/Memberships/MembersList.tsx

```diff
@@ -13,9 +17,23 @@ export const MembersList = <T extends TargetId>({
 	targetId,
 	readOnly,
 }: MembersListProps<T>) => {
+	const dedupedMembers = useMemo(() => {
+		const dedupedMembers = new Map<UsersId, MembersListProps<T>["members"][number]>();
+		for (const member of members) {
+			if (!dedupedMembers.has(member.id)) {
+				dedupedMembers.set(member.id, member);
+			} else {
+				const m = dedupedMembers.get(member.id);
+				if (m && firstRoleIsHigher(member.role, m.role)) {
+					dedupedMembers.set(member.id, m);
+				}
+			}
+		}
+		return dedupedMembers;
+	}, [members]);
 	return (
 		<>
-			{members.map((user) => (
+			{[...dedupedMembers.values()].map((user) => (
```

Comment:
not a blocker as i think we should just get this in, but i think it would prefer it if we just returned a single membership from the db with an optional `forms` property holding the forms the user has access to. i think that's nicer than having to do something like this on the client! but let's just get this in first, we can do that when we get to it (unless ofc you think we shouldnt do it at all)

Reply:
I agree, this is ugly and a temporary fix. How about you review my PR for #1138 when it's up and see if you still want me to change the query?

Reply:
actually what i really mean is: that sounds like it's probably a good idea and i will most likely do it in #1138 😆